Introduce assembler pipelines and identify pipelines

gicmo opened the pull request:
This converts the assembler phase of the `Pipeline`[1] with a new pipeline that has the previous assembler as the first and only stage. This was possible to do since the `Assembler` previously got converted into a `Stage`, and that stage has the tree as `Input`; so this stage just got moved to a newly created `Pipeline`. 
Additionally, all pipelines in the manifest now have to have a name assigned. This is currently used in the version 1 format code (`formats/v1`) to identify the pipeline that builds the tree and the pipeline that corresponds to the assembler. 
The inputs of a stage are now included in the `id` calculation. NB: This, and the move of the assembler to a new pipeline, changes the `id` of the assembler.

[1] That is the main pipeline, since for build pipelines the assembler is not run, even if defined.

Replacs #554

teg commented on osbuild/pipeline.py:
```diff
@@ -119,7 +119,7 @@ def __init__(self, runner=None, build=None):
     @property
     def id(self):
-        return self.output_id or self.tree_id
+        return self.output_id or self.tree_id or id(self)
```
I think it would be helpful to have a comment on the `id` property. What is it meant to be used for / what properties is it required to have?

teg replied:
This was a general musing, could be added (if you agree) in a follow-up.

teg reviewed (dismissed):
Love it! Can't find anything to nitpick.

gicmo commented:
I will actually re-work the `id` commits.

gicmo commented:
I have updated the PR to completely remove `tree_id` usages. Also instead of having `build-N` for nested build pipelines I am using `build-build-build...` now because I think that makes the nesting clearer (not that anyone would ever actually see that). I added more tests, esp. for the nesting case.

achilleas-k commented on osbuild/pipeline.py:
```diff
@@ -330,16 +334,22 @@ def mark_pipeline(pl):
+            if pl.id == name_or_id:
                 return pl
-        raise KeyError("{pipeline_id} not found")
+        raise KeyError("{name_or_id} not found")
```
Missing f-string prefix

gicmo replied:
Fixed in a later commit. I might have forgotten to mention that in the corresponding later commit though.

achilleas-k replied:
oh, didn't notice the [Outdated], I usually look out for those.

teg reviewed (approved):
So good!

achilleas-k reviewed (commented):
Found the smallest issue.
